File: PlacementCellBackend/Services/Analytics/Interfaces/IPlacementAnalyticsService.cs
using PlacementCellBackend.DTOs;

namespace PlacementCellBackend.Services.Analytics.Interfaces
{
    public interface IPlacementAnalyticsService
    {
        // Dashboard & Overview
        Task<DashboardStats> GetDashboardStatsAsync();

        // Company Analytics
        Task<IEnumerable<CompanyRanking>> GetTopCompaniesRankingAsync(int topN = 10);
        Task<CompanyInsights> GetCompanyInsightsAsync(string companyId);

        // Alumni Analytics
        Task<AlumniStatistics> GetAlumniStatisticsAsync();
        Task<IEnumerable<RecentPlacement>> GetRecentPlacementsAsync(int count = 10);

        // Placement Trends
        Task<PlacementTrends> GetPlacementTrendsAsync();

        // Resources Analytics
        Task<RecommendedResources> GetTopRecommendedResourcesAsync(int topN = 10);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Analytics/Interfaces/IPlacementAnalyticsService.java
package com.example.placementicsbackend.services.analytics.interfaces;

import com.example.placementicsbackend.dtos.AlumniStatistics;
import com.example.placementicsbackend.dtos.CompanyInsights;
import com.example.placementicsbackend.dtos.CompanyRanking;
import com.example.placementicsbackend.dtos.DashboardStats;
import com.example.placementicsbackend.dtos.PlacementTrends;
import com.example.placementicsbackend.dtos.RecentPlacement;
import com.example.placementicsbackend.dtos.RecommendedResources;
import java.util.List;

public interface IPlacementAnalyticsService {

    DashboardStats getDashboardStats();

    List<CompanyRanking> getTopCompaniesRanking(int topN);

    CompanyInsights getCompanyInsights(String companyId);

    AlumniStatistics getAlumniStatistics();

    List<RecentPlacement> getRecentPlacements(int count);

    PlacementTrends getPlacementTrends();

    RecommendedResources getTopRecommendedResources(int topN);
}
